### DISCUSSION TITLE
How to use the executable created from Pyinstaller import new things, for example another python module that are on the PYTHONPATH?

### DISCUSSION THREAD

**Discussion by KarenMars** [pyinstaller/pyinstaller#8225] [Help] [answered]
Hi,

In the [FAQ](https://github.com/pyinstaller/pyinstaller/wiki/FAQ), it mentions that the executable created from Pyinstaller can import new things that are on PYTHONPATH. Does it mean that we can import a new python module when running the Pyinstaller executable? Is there any guidelines for this issure? Thank you very much.

**Comment by rokm** [comment, marked as answer]
Like the FAQ states, `PYTHONPATH` is ignored, so you cannot use it to extend the module search path for PyInstaller-frozen application. But you can (programmatically, at the start of your program) extend `sys.path` to load modules from external locations.

But to do so, you will need to ensure yourself that external modules' dependencies are available, and that none of those dependencies are partially collected into the frozen application (if you plan to provide a full copy in external location). See https://github.com/orgs/pyinstaller/discussions/8190

**Reply by KarenMars** [reply]
Thank you very much, the answer in the link is very clear.
